File: 01_Array_and_String/Python Code/04_605. Can Place Flowers.py
# Problem Link -> https://leetcode.com/problems/can-place-flowers/description/

class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        length = len(flowerbed)
        if n == 0:
            return True
        
        for i in range(length):
            if flowerbed[i] == 0:
                checkLeftEmpty = (i == 0) or (flowerbed[i - 1] == 0)
                checkRightEmpty = (i == length - 1) or (flowerbed[i + 1] == 0)

                if(checkLeftEmpty and checkRightEmpty):
                    flowerbed[i] = 1
                    n -= 1

                    if n == 0:
                        return True
        
        return False



# Time Complexity -> O(n)
# Space Complexity -> O(1)

File: 01_Array_and_String/C++ Code/04_605. Can Place Flowers.cpp
// Problem Link -> https://leetcode.com/problems/can-place-flowers/description/

class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        int len = flowerbed.size();
        if(n == 0) return true;

        for(int i = 0; i < len; i++){
            if(flowerbed[i] == 0){
                bool checkLeftEmpty = (i == 0) || (flowerbed[i - 1] == 0);
                bool checkRightEmpty = (i == len - 1) || (flowerbed[i + 1] == 0);

                if(checkLeftEmpty && checkRightEmpty){
                    flowerbed[i] = 1;
                    n--;

                    if(n == 0) return true;
                }
            }
        }

        return false;
    }
};


// Time Complexity -> O(n)
// Space Complexity -> O(1)
